libs/gamify/applications/src/lib/dto/create-application.input.ts

```diff
@@ -0,0 +1,24 @@
+import { ApplicationType } from '@gamify/core';
+import { IsEnum, IsNotEmpty, IsNumber, IsOptional, IsUrl } from 'class-validator';
+
+export class CreateApplicationInput {
+
+    @IsNotEmpty()
+    name: string;
+
+    @IsOptional()
+    description: string;
+
+    @IsEnum(ApplicationType)
+    applicationType: string;
+
+    @IsOptional()
+    @IsUrl()
+    externalApplicationUrl: string;
+
+    // Admin user id should not be provided by the client as input value but 
+    // should be obtained from current authenticated user.
+    @IsOptional()
+    @IsNumber()
+    adminUserId: number;
```

Comment:
I'm not in favour of having a property named `adminUserId`. How is this different than just `userId`?

Reply:
Changed it to ownerUserId for clarification